### PR TITLE
Add custom warning to Tools

### PR DESCRIPTION
**Pull request by micahstairs**
One of the biggest requests from players is to improve the UX with Tools (https://boardgamearena.com/bug?id=18164). this is my proposal for addressing it.
Screen Shot 2022-09-03 at 11 51 27 AM

Please note that I structured the JS code in a way that it will make it easy to add confirmation buttons on a timer like we do for meld/dogma actions. I can easily see us wanting to give players an option to confirm card selections as well.

### CODE REVIEW

**Review by lordmcfuzz — APPROVED** [review]

**Review by ultimatefiend — COMMENTED** [review]
The confirm of the "3" specifically is a good thing to do.  Does this prompt on all three returns or just the first?  

It seems a bit wordy.  Do we really need the second part of the explanation?  Alternatively, it could state what won't happen (they won't draw 3 ones if the 3 is returned in the first dogma effect).  That would make the reason for the prompt clearer.

**Comment by micahstairs** [comment]
It only prompts on the return of the first card (if it's a 3). This is why the backend is passing that `num_cards_already_chosen` variable now, so that we can tell if it's the first card being returned or not.

I agree that it's pretty wordy. I'm definitely open to suggestions on new wordings!

**Comment by micahstairs** [comment]
How about `Are you sure you want to return a 3 as part of the first non-demand effect? This won't allow you to draw three cards.`?

**Comment by micahstairs** [comment]
This is what it looks like now:
Screen Shot 2022-09-08 at 8 49 17 PM

**Comment by micahstairs** [comment]
I shortened the warning message even more:

Screen Shot 2022-09-08 at 8 54 24 PM
